### DISCUSSION TITLE
t5l1 dgus2 support

### DISCUSSION THREAD

**Discussion by damian5602** [Desuuuu/klipper#122] [General]
Hi, i got my flsun SR with t5l1 dgus2. Is there any chance to get it working? After uploading t5uid1 files it does not see most files.
![281653029_582255439728283_2921276382108366742_n](https://user-images.githubusercontent.com/6631231/169689761-56ff0175-26ff-4905-b8f8-1bc73ad711bc.jpg)

Sorry if it's not a good place for this

**Comment by Desuuuu** [comment]
I don't have time to work on this personally.

Also, I would highly recommend not flashing your touchscreen with incompatible files (unless you're fine bricking it).

**Comment by ratomas108** [comment]
Hello,
try this FW modified for T5L1 display:
https://github.com/Desuuuu/DGUS-reloaded-Klipper/discussions/33

**Reply by fryc88** [reply]
Hi,
I have Sovol SV01 PRO with screen based on T5L1, however it's been installed vertical there, but that's not an issue.
When i try to flash your firmware, it doesn't really show anything, just got half of screen with part of blueish logo and that's it. No signs of klipper there, and during the flashing i think it doesn't take all files there.
Any ideas? I wish i could use this screen with klipper somehow, but Sovol did not release source code and i can't find anything to try to modify it in Dgwin Dgus, in fact, i can't even find version mentioned by you (8.x), only 7.x.

![IMG-8448](https://user-images.githubusercontent.com/50572708/201908485-7b8f96f8-0807-4204-9eb8-c091811ed88d.jpg)

![IMG-8447](https://user-images.githubusercontent.com/50572708/201908474-ad72684c-e84a-483e-bbe1-920934084a91.jpg)

**Comment by ratomas108** [comment]
Hi,
I used DGUS_Tool_V8.2.1.14, but here is DGUS_V7641 for T5L display: https://www.dwin-global.com/tool/. I opened DWprj in DGUS Tool. I used original T5L.cfg, HZK file, generated new bin files, recompiled icons, buttons, screens. Try these tutorials: https://www.dwin-global.com/t5l-dgus-v7-6-video-tutorial/. I compared Creality and Desuuuu FW.
Creality files for Sermoon:
![image](https://user-images.githubusercontent.com/106281051/201928397-345a6576-e6a1-4c14-990e-03014b52373e.png)
You have to recompile:
24_icons, 27_buttons, 30_progress_left, 37_progress_right, screens
![image](https://user-images.githubusercontent.com/106281051/201929273-612e2901-4801-433b-b7a0-20c702e76cb4.png)
My FW:
![image](https://user-images.githubusercontent.com/106281051/201931497-5e21eb3f-e080-4675-8d23-204ebe1ae32c.png)
0_DWIN_ASC.HZK and T5LCFG_480272.CFG are files from Creality FW, others are generated in DGUS Tool.

**Reply by fryc88** [reply]
Have you got any idea why my screen might be stuck like this?
I'm assuming it supposed to show some Klipper logo and then show everything else once communication is working?
I'm a bit stuck with it and i'm not that great with compiling everything from scratch, etc., maybe you could provide your config files so i could have a look at it?

**Reply by ratomas108** [reply]
I think you need *.HZK and *.CFG files for your display.

**Comment by fryc88** [comment]
I have them only from marlin built firmware for my printer, as I said they didn't release source for the screen. can those be used to add to project too? I've found Ender 6 I think source code for this type of vertical screen, but I also know there was few types of them looking similar.
Can i possibly contact you somewhere directly to ask for few advises?

**Reply by ratomas108** [reply]
I used source code from Desuuuu and FW from Creality ( Marlin ). I have no source code for my display. Open your *.CFG file in project ( Setting\DGUS ).
![image](https://user-images.githubusercontent.com/106281051/201952551-994f5f98-9db8-4826-8a6b-43b9a2c071d5.png)
Use your *.CFG and *.HZK:
![image](https://user-images.githubusercontent.com/106281051/201954923-7e186d99-d002-4e00-8bd4-20625bc43573.png)

**Reply by fryc88** [reply]
And where/how do you load HZK file?

**Reply by ratomas108** [reply]
I used HZK file only for flashing FW ( without loading to project ).
Here are some steps about how to flash FW.
https://github.com/Desuuuu/DGUS-reloaded-Klipper/discussions/26

**Comment by Thinkersbluff** [comment]
FWIW, I have now refactored the DGUS-Reloaded_Klipper UI to run on any version of DGUS2 on the T5L 272x480 displays that come stock with Creality CR6 printers.

My DWIN_SET project is posted on the default branch of this repo: https://github.com/Thinkersbluff/DGUS-reloadedForKlipper_CR6/tree/Refactored_For_CR6_T5L_DWIN_Display
I have not included either a T5L.cfg file nor an HZK file, assuming that users will have already flashed those files to their displays in the past.
